src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        stateCopy = { ...stateCopy, ...extraData };
+        break;
+
+      case 'removeProperties':
+        for (const item of keysToRemove) {
+          delete stateCopy[item];
+        }
+        break;
+
+      case 'clear':
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        break;
+    }
```

Comment:
The switch statement is missing a default case. It's a good practice to include a default case for error handling, even if it's just to throw an error or to handle unexpected `type` values gracefully.